src/Microsoft.DocAsCode.Plugins/IHostService.cs

```diff
@@ -20,7 +20,32 @@ public interface IHostService
         ImmutableList<FileModel> GetModels(DocumentType? type = null);
         ImmutableList<FileModel> LookupByUid(string uid);
 
-        void ReportDependency(FileModel currentFileModel, ImmutableArray<string> dependency);
+        /// <summary>
+        /// report dependency
+        /// </summary>
+        /// <param name="from">'from' node's file path from working directory</param>
+        /// <param name="to">'to' node's file path from working directory</param>
+        /// <param name="reportedBy">'reportedby' node's file path from working directory</param>
+        /// <param name="type">dependency type</param>
+        void ReportDependency(string from, string to, string reportedBy, string type);
+
+        /// <summary>
+        /// report dependency to
+        /// </summary>
+        /// <param name="currentFileModel">filemodel of 'from' node</param>
+        /// <param name="to">'to' node's file path from working directory or file path relative to 'from' filemodel</param>
+        /// <param name="type">dependency type</param>
+        void ReportDependencyTo(FileModel currentFileModel, string to, string type);
```

Comment:
okay...

---
In reply to: [78131566](https://github.com/dotnet/docfx/pull/681#discussion_r78131566) [](ancestors = 78131566,78126216)